tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -1411,21 +1413,23 @@ - (void)dealloc {
         __block auto uxEvent = UXEvent::CreateAuto();
         __block auto xamlSubscriber = std::make_shared<XamlEventSubscription>();
         __block NSString* expectedBackgroundImage = @"150x150.png";
+        EventRegistrationToken ert{};
+        EventRegistrationToken* const ertAddr = &ert;
 
         WXFrameworkElement* xamlElement = [buttonToTest xamlElement];
         ASSERT_OBJCNE(xamlElement, nil);
 
         // Wait for the layerContent to be part of the visual tree
         dispatch_sync(dispatch_get_main_queue(), ^{
-            EventRegistrationToken ert = [xamlElement addLayoutUpdatedEvent:^(RTObject* sender, RTObject* args) {
+            *ertAddr = [xamlElement addLayoutUpdatedEvent:^(RTObject* sender, RTObject* args) {
```

Comment:
I've verified that things _should_ work as you describe. I believe the reason it works with a pointer and not with a __block variable is because of a compiler bug, which I'll follow up on.